src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -210,6 +228,70 @@ private boolean canGoPlanting(@NotNull Field currentField, @NotNull BuildingFarm
         return true;
     }
 
+    private AIState cycle()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "cycle" is 18 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)